torchreid/integration/nncf/accuracy_aware_training/common/runner.py

```diff
@@ -0,0 +1,327 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Union
+from typing import Tuple
+from typing import TypeVar
+from abc import ABC
+from abc import abstractmethod
+import pathlib
+from nncf.api.compression import CompressionAlgorithmController
+from nncf.common.utils.backend import infer_backend_from_compression_controller
+from nncf.common.utils.backend import BackendType
+
+ModelType = TypeVar('ModelType')
+OptimizerType = TypeVar('OptimizerType')
+LRSchedulerType = TypeVar('LRSchedulerType')
+TensorboardWriterType = TypeVar('TensorboardWriterType')
+
+
+class TrainingRunner(ABC):
```

Comment:
Are this train runner abstract class and other abstract classes unique for d-o-r? Seems like this stuff could be located somewhere inside nncf package.

Reply:
done